File: refactor.js
const fs = require('fs');
const path = require('path');

const stores = {
  useUIStore: [
    'currentPage',
    'setCurrentPage',
    'helpTab',
    'setHelpTab',
    'sessionsFocusDate',
    'setSessionsFocusDate',
    'clearSessionsFocusDate',
    'sessionsFocusRange',
    'setSessionsFocusRange',
    'sessionsFocusProject',
    'setSessionsFocusProject',
    'projectPageId',
    'setProjectPageId',
    'firstRun',
    'setFirstRun',
  ],
  useDataStore: [
    'dateRange',
    'timePreset',
    'setDateRange',
    'setTimePreset',
    'shiftDateRange',
    'canShiftForward',
    'refreshKey',
    'triggerRefresh',
    'autoImportDone',
    'autoImportResult',
    'setAutoImportDone',
  ],
  useSettingsStore: [
    'currencyCode',
    'setCurrencyCode',
    'chartAnimations',
    'setChartAnimations',
  ],
};

function getStore(prop) {
  for (const [store, props] of Object.entries(stores)) {
    if (props.includes(prop)) return store;
  }
  return 'useAppStore'; // fallback
}

function processFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf-8');
  if (!content.includes('useAppStore')) return;

  const originalContent = content;
  const importsToAdd = new Set();

  // 1. replace: const { foo, bar } = useAppStore();
  content = content.replace(
    /const\s+\{\s*([^}]+)\s*\}\s*=\s*useAppStore\(\);/g,
    (match, propsStr) => {
      const props = propsStr
        .split(',')
        .map((p) => p.trim())
        .filter(Boolean);
      const storeMap = {};
      for (const p of props) {
        const cleanP = p.split(':')[0].trim(); // handle aliases
        const s = getStore(cleanP);
        if (!storeMap[s]) storeMap[s] = [];
        storeMap[s].push(p);
        importsToAdd.add(s);
      }
      return Object.entries(storeMap)
        .map(([s, pList]) => `const { ${pList.join(', ')} } = ${s}();`)
        .join('\n  ');
    },
  );

  // 2. replace: useAppStore((s) => s.prop)
  content = content.replace(
    /useAppStore(?:<.*?>)?\(\s*\([^)]*\)\s*=>\s*[a-zA-Z_]\.([a-zA-Z0-9_]+)\)/g,
    (match, prop) => {
      const s = getStore(prop);
      importsToAdd.add(s);
      return `${s}((s) => s.${prop})`;
    },
  );

  // 3. replace useAppStore.getState().prop
  content = content.replace(
    /useAppStore\.getState\(\)\.([a-zA-Z0-9_]+)/g,
    (match, prop) => {
      const s = getStore(prop);
      importsToAdd.add(s);
      return `${s}.getState().${prop}`;
    },
  );

  // 4. Update imports
  if (importsToAdd.size > 0) {
    const importStatements = Array.from(importsToAdd).map((s) => {
      if (s === 'useUIStore')
        return "import { useUIStore } from '@/store/ui-store';";
      if (s === 'useDataStore')
        return "import { useDataStore } from '@/store/data-store';";
      if (s === 'useSettingsStore')
        return "import { useSettingsStore } from '@/store/settings-store';";
    });
    content = content.replace(
      /import\s+\{\s*useAppStore\s*\}\s*from\s*['"]@\/store\/app-store['"];?/g,
      importStatements.join('\n'),
    );
  }

  if (content !== originalContent) {
    fs.writeFileSync(filePath, content, 'utf-8');
    console.log(`Updated ${filePath}`);
  }
}

function walk(dir) {
  const list = fs.readdirSync(dir);
  for (const file of list) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    if (stat && stat.isDirectory()) {
      walk(filePath);
    } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
      processFile(filePath);
    }
  }
}

walk('c:/_cloud/__cfab_demon/__client/dashboard/src');

File: refactor.py
import os, sys, re

stores = {
    'useUIStore': ['currentPage', 'setCurrentPage', 'helpTab', 'setHelpTab', 'sessionsFocusDate', 'setSessionsFocusDate', 'clearSessionsFocusDate', 'sessionsFocusRange', 'setSessionsFocusRange', 'sessionsFocusProject', 'setSessionsFocusProject', 'projectPageId', 'setProjectPageId', 'firstRun', 'setFirstRun'],
    'useDataStore': ['dateRange', 'timePreset', 'setDateRange', 'setTimePreset', 'shiftDateRange', 'canShiftForward', 'refreshKey', 'triggerRefresh', 'autoImportDone', 'autoImportResult', 'setAutoImportDone'],
    'useSettingsStore': ['currencyCode', 'setCurrencyCode', 'chartAnimations', 'setChartAnimations']
}

def get_store(prop):
    for store_name, props in stores.items():
        if prop in props: return store_name
    return 'useAppStore'

def process_file(path):
    try:
        with open(path, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        print(f"Error reading {path}: {e}")
        return

    if 'useAppStore' not in content:
        return

    original_content = content
    imports_to_add = set()

    # 1. replace const { ... } = useAppStore();
    def repl_destructure(m):
        props_str = m.group(1)
        props = [p.strip() for p in props_str.split(',') if p.strip()]
        
        store_map = {}
        for p in props:
            # Handle aliases e.g. "triggerRefresh: refresh" -> "triggerRefresh"
            clean_p = p.split(':')[0].strip()
            s = get_store(clean_p)
            if s not in store_map:
                store_map[s] = []
            store_map[s].append(p)
            imports_to_add.add(s)
            
        res = []
        for s, p_list in store_map.items():
            res.append(f"const {{ {', '.join(p_list)} }} = {s}();")
        return "\n  ".join(res)

    content = re.sub(r'const\s+\{\s*([^}]+)\s*\}\s*=\s*useAppStore\s*\(\s*\)\s*;', repl_destructure, content)

    # 2. replace useAppStore((s) => s.prop)
    def repl_selector(m):
        prop = m.group(1)
        s = get_store(prop)
        imports_to_add.add(s)
        return f"{s}((s) => s.{prop})"
    content = re.sub(r'useAppStore\s*\(\s*\(\s*[a-zA-Z_]\s*\)\s*=>\s*[a-zA-Z_]\.([a-zA-Z0-9_]+)\s*\)', repl_selector, content)
    
    # 3. replace multi-line useAppStore((s) => { return s.prop })
    def repl_selector_block(m):
        prop = m.group(1)
        s = get_store(prop)
        imports_to_add.add(s)
        # We simplify it to inline here for simplicity unless it has logic, assuming simple returns
        return f"{s}((s) => s.{prop})"
    content = re.sub(r'useAppStore\s*\(\s*\(\s*[a-zA-Z_]\s*\)\s*=>\s*\{\s*return\s*[a-zA-Z_]\.([a-zA-Z0-9_]+);?\s*\}\s*\)', repl_selector_block, content)

    # 4. replace useAppStore.getState().prop
    def repl_getstate(m):
        prop = m.group(1)
        s = get_store(prop)
        imports_to_add.add(s)
        return f"{s}.getState().{prop}"
    content = re.sub(r'useAppStore\.getState\(\)\.([a-zA-Z0-9_]+)', repl_getstate, content)

    # 5. Add imports
    if imports_to_add:
        # replace original import
        import_strs = []
        for s in imports_to_add:
            if s == 'useUIStore': import_strs.append("import { useUIStore } from '@/store/ui-store';")
            if s == 'useDataStore': import_strs.append("import { useDataStore } from '@/store/data-store';")
            if s == 'useSettingsStore': import_strs.append("import { useSettingsStore } from '@/store/settings-store';")
        content = re.sub(r'import\s+\{[^}]*useAppStore[^}]*\}\s+from\s+[\'"]@/store/app-store[\'"];?', '\n'.join(import_strs), content)

    if content != original_content:
        with open(path, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated: {path}")
    else:
        print(f"Matched 'useAppStore' but regexes didn't change: {path}")

for root, dirs, files in os.walk('c:/_cloud/__cfab_demon/__client/dashboard/src'):
    for file in files:
        if file.endswith(('.ts', '.tsx')) and file != 'app-store.ts':
            process_file(os.path.join(root, file))
